src/roles/dullahan.py

```diff
@@ -59,16 +59,16 @@ def on_player_win(evt, var, user, role, winner, survived):
         evt.data["iwon"] = True
 
 @event_listener("del_player")
-def on_del_player(evt, cli, var, nick, mainrole, allroles, death_triggers):
+def on_del_player(evt, var, user, mainrole, allroles, death_triggers):
     for h, v in list(KILLS.items()):
-        if v.nick == nick:
+        if v is user:
             h.send(messages["hunter_discard"])
             del KILLS[h]
-        elif h.nick == nick:
+        elif h is user:
             del KILLS[h]
     if death_triggers and "dullahan" in allroles:
         pl = evt.data["pl"]
```

Comment:
evt.data["pl"] should be updated to be a user list as well (this will require modifications in wolfgame.py as well in del_player and refresh_pl)